testdata/var_declarations_zero_value.go

```diff
@@ -0,0 +1,9 @@
+package fixtures
+
+func zeroValueExample() {
+	var y int                // No warning, zero value is implicit
+	var z1 any = 0           // No warning, zero value for any is nil
```

Comment:
IMO, in cases of the form `var <id> any = <value>` where value is not `nil` the rule must warn with `should omit type any from declaration of var <id>; it will be inferred from the right-hand side` because `var <id> = <value>` is equivalent

Reply:
Yes

https://go.dev/play/p/TORLftc7jlR with any
https://go.dev/play/p/dYvM0OlUstA without any

Reply:
if an initial value is provided then declaring the variable of being type `any` has no utility

Reply:
Still, I think there is a difference. With `any` it is a dynamic type and it can be changed to any value later. Consider this:

```
package main

func main() {
	var a1 = 1
	// a1 = "string" // fail to compile
	var a2 any = 1
	a2 = "string" // this will compile
	_ = a1
	_ = a2
}
```

Reply:
> declaring the variable of being type `any` has no utility

I believe there can be a reason to declare an `any` type variable in case if you plan to have an arbitrary value for it later.

Reply:
Ooops; you are right... I  did not think in these kind of (ugly) cases